experimental/PropertyDDS/packages/property-dds/src/test/propertyTree.spec.ts

```diff
@@ -196,6 +173,102 @@ describe("PropertyDDS summarizer", () => {
 
 		expect(u1.root.get<ArrayProperty>(USERS)?.getValues().length).to.equal(1);
 	});
+
+	it("Scenario 2 (repeated summarization)", async function () {
+		/**
+		 * This test produces a scenario where we have an empty remoteChanges array in the summarizer client
+		 * and then get more changes that cannot yet be pruned away, because the MSN continues to point to the
+		 * previous head commit.
+		 *
+		 * We used to have a bug that was caused by this, where the prune code would prune the remote changes
+		 * but not the unrebased remote changes, causing rebase errors.
+		 */
+		this.timeout(30000);
+
+		// 1- U1 joins together with summarizer
+		const {
+			client: u1,
+			summarizer,
+			dataObject: dataObject1,
+			container: container1,
+		} = await getClient(true);
+		await objProvider.ensureSynchronized();
+
+		// 2- Make some modifications
+		u1.root.insert("c1", PropertyFactory.create("NodeProperty"));
+		u1.commit();
+		await objProvider.ensureSynchronized();
+
+		u1.root.insert("c2", PropertyFactory.create("NodeProperty"));
+		u1.commit();
+
+		await objProvider.ensureSynchronized();
+
+		const {
+			client: u2,
+			dataObject: dataObject2,
+			container: container2,
+		} = await getClient(false, true);
+		await objProvider.ensureSynchronized();
+
+		// We do two changes to a different DDS (the root map), to make sure, that
+		// updates are triggered that do not affect the propertyDDS
+		dataObject1.root.set("c2", "aaa");
+		dataObject2.root.set("c2", "aaa");
+		await objProvider.ensureSynchronized();
+
+		// Now we wait until the msn has sufficiently advanced that the pruning below
+		// will remove all remoteChanges
+		await synchronizeMSN(container2, container1);
+
+		// Summarize
+		await summarizeNow(summarizer.summarizer);
+
+		const summarizerDataObject = await requestFluidObject<ITestFluidObject>(
+			summarizer.container,
+			"/",
+		);
+		const summarizerClient =
+			await summarizerDataObject.getSharedObject<SharedPropertyTree>(propertyDdsId);
+
+		// Make changes only on u1, u2 must not advance to make sure
+		// the msn is not advanced
+		u1.root.insert("a", PropertyFactory.create("NodeProperty"));
+		u1.commit();
+
+		u1.root.insert("b", PropertyFactory.create("NodeProperty"));
+		u1.commit();
+
+		await objProvider.opProcessingController.processOutgoing(container1);
+		await objProvider.opProcessingController.processIncoming(container1);
+
+		// Summarize again
+		await summarizeNow(summarizer.summarizer);
+
+		// Make sure the summarizer did not delete any of the unrebased changes
+		expect(summarizerClient.remoteChanges.length).to.equal(2);
+		expect(Object.keys(summarizerClient.unrebasedRemoteChanges).length).to.equal(2);
+	});
+
+	async function synchronizeMSN(container2: IContainer, container1: IContainer) {
```

Comment:
NIT: Its cleaner and more readable if we move this helper method to the top of the file